pyomo/solvers/plugins/solvers/mosek_direct.py

```diff
@@ -361,6 +362,12 @@ def _add_vars(self, var_seq):
         self._pyomo_var_to_solver_var_map.update(zip(var_seq, var_ids))
         self._solver_var_to_pyomo_var_map.update(zip(var_ids, var_seq))
         self._referenced_variables.update(zip(var_seq, [0]*len(var_seq)))
+        # Initialize variables
+        for m_v, p_v in zip(var_ids, var_seq):
+            if p_v.value is not None:
+                for sol_type in mosek.soltype.values:
+                    self._solver_model.putxxslice(
+                        sol_type, m_v, m_v + 1, [(p_v.value)])
```

Comment:
Question: by moving this to `_add_vars()`, does this prevent the variable values from being updated between subsequent solves in the case of the persistent solver?

Reply:
The same `task` object instance will be used over subsequent solves (when using mosek_persistent). So, the variable values from the previous solve will already be in that particular task. Pyomo would not have to pass them again. We only need to tell mosek if it should use those values or ignore them.

Reply:
@Utkarsh-Detha, reminder that this PR is waiting on a response from you to the above comment from @jsiirola.

Reply:
@jsiirola then I suppose we just leave it as is. I will undo the changes concerning this.

(Sorry about the slow response, I have been on holiday)